src/lib.rs

```diff
@@ -137,126 +169,113 @@ pub struct OfferInfo {
 }
 
 pub struct LightningNode {
-    core_node: Arc<eel::LightningNode>,
+    sdk: Arc<BreezServices>,
     auth: Arc<Auth>,
     fiat_topup_client: PocketClient,
     offer_manager: OfferManager,
+    rt: Runtime,
+}
+
+struct LipaEventListener;
+
+impl EventListener for LipaEventListener {
+    fn on_event(&self, e: BreezEvent) {
+        match e {
+            BreezEvent::NewBlock { .. } => {}
+            BreezEvent::InvoicePaid { .. } => {}
+            BreezEvent::Synced => {}
+            BreezEvent::PaymentSucceed { .. } => {}
+            BreezEvent::PaymentFailed { .. } => {}
+            BreezEvent::BackupStarted => {}
+            BreezEvent::BackupSucceeded => {}
+            BreezEvent::BackupFailed { .. } => {}
+        }
+    }
 }
 
 impl LightningNode {
     pub fn new(config: Config, events_callback: Box<dyn EventsCallback>) -> Result<Self> {
         enable_backtrace();
         fs::create_dir_all(&config.local_persistence_path).map_to_permanent_failure(format!(
             "Failed to create directory: {}",
-            config.local_persistence_path,
+            &config.local_persistence_path,
         ))?;
         if config.enable_file_logging {
             init_logger_once(
                 LOG_LEVEL,
                 &Path::new(&config.local_persistence_path).join(LOGS_DIR),
             )?;
         }
-        let seed = sanitize_input::strong_type_seed(&config.seed)
-            .map_runtime_error_using(RuntimeErrorCode::from_eel_runtime_error_code)?;
+
+        let rt = Builder::new_multi_thread()
+            .worker_threads(4)
+            .thread_name("breez-async-runtime")
```

Comment:
I'm not sure this will only be used for Breez tasks. Maybe we can rename it to "3L-async-runtime"?